cognite/client/_api_client.py

```diff
@@ -336,12 +342,51 @@ def _list(
             limit=limit,
             chunk_size=self._LIST_LIMIT,
             filter=filter,
-            headers=headers,
             other_params=other_params,
+            headers=headers,
         ):
             items.extend(resource_list.data)
         return cls(items, cognite_client=self._cognite_client)
 
+    def _list_partitioned(
+        self,
+        partitions,
+        cls=None,
+        resource_path: str = None,
+        filter: Dict = None,
+        other_params={},
+        headers: Dict = None,
+    ):
+        cls = cls or self._LIST_CLASS
+        resource_path = resource_path or self._RESOURCE_PATH
+
+        def get_partition(partition):
+            next_cursor = None
+            retrieved_items = []
+            while True:
+                body = {
+                    "filter": filter or {},
+                    "limit": self._LIST_LIMIT,
+                    "cursor": next_cursor,
+                    "partition": partition,
+                    **other_params,
+                }
+                res = self._post(url_path=resource_path + "/list", json=body, headers=headers)
+                retrieved_items.extend(res.json()["items"])
+                next_cursor = res.json().get("nextCursor")
+                if next_cursor is None:
+                    break
+            return cls._load(retrieved_items, cognite_client=self._cognite_client)
+
+        tasks = [("{}/{}".format(i + 1, partitions),) for i in range(partitions)]
+        tasks_summary = utils._concurrency.execute_tasks_concurrently(get_partition, tasks, max_workers=partitions)
+        if tasks_summary.exceptions:
+            raise tasks_summary.exceptions[0]
+        result = tasks_summary.results[0]
```

Comment:
return tasks_summary.joined_results()

Reply:
I think that will give a list, not an AssetList
joined_results = []  etc